theano/sandbox/cuda/basic_ops.py

```diff
@@ -3329,20 +3329,42 @@ def c_code_cache_version(self):
         return ()
 
 
-class GpuFlatten(gof.HideC, tensor.Flatten, GpuOp):
+class GpuFlatten(gof.HideC, tensor.Reshape, GpuOp):
     """
     Implement Flatten on the gpu.
 
+    Note: The interface GpuFlatten is deprecated, you should use gpu_flatten
     """
 
     def make_node(self, x):
         assert isinstance(x.type, CudaNdarrayType)
-        rval = tensor.Flatten.make_node(self, x)
+        rval = flatten(x)
         host_out_broadcastable = rval.outputs[0].type.broadcastable
         out_type = CudaNdarrayType(broadcastable=host_out_broadcastable)
         return Apply(self, [x], [out_type()])
 
 
+
+def gpu_flatten(x, outdim=1):
+    """
+    Implement flatten on the gpu.
+
+    :param x: the variable that should be reshaped.
+    :type x: theano.tensor.var.TensorVariable
+
+    :param outdim: the number of dimensions of the returned variable
+    :type outdim: int
+
+    :returns: the flattend variable with dimensionality of outdim
```

Comment:
This is the old docstring format. Please follow this [guide](https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt)